File: beecrowd/1023-drought/py/main.py
# This is giving time limit exceeded

import math
from sys import stdin, stdout

class Property:
    def __init__(self, residents = 0, consumption = 0) -> None:
        self.residents = residents
        self.consumption = consumption
        self.avg = 0
    
    @staticmethod
    def printCityStatistics(properties, city_number):
        stdout.write("Cidade# {}:\n".format(city_number))

        property_stats = ""
        sum_consumption = 0
        sum_residents = 0

        for property in properties:
            property_stats += "{}-{} ".format(property.residents, property.avg)
            sum_consumption += property.consumption
            sum_residents += property.residents
        property_stats = property_stats[:-1] + '\n'

        stdout.write(property_stats)
        stdout.write("Consumo medio: {:.2f} m3.\n".format(math.floor(sum_consumption/sum_residents * 100)/100.0))

city_number = 1

while True:
    nb_properties = int(stdin.readline())

    if nb_properties < 1:
        break
    
    if city_number > 1:
        stdout.write('\n')

    properties = []

    for i in range(nb_properties):
        property = Property()
        property.residents, property.consumption  = map(int, stdin.readline().split())
        property.avg = property.residents // property.consumption
        properties.append(property)
    properties.sort(key=lambda property: property.avg)
    
    Property.printCityStatistics(properties, city_number)

    city_number += 1

File: beecrowd/1023-drought/go/main.go
package main

import (
	"fmt"
	"math"
	"sort"
)

type Property struct {
	Residents   int
	Consumption int
}

func NewProperty(residents, consumption int) *Property {
	return &Property{
		Residents:   residents,
		Consumption: consumption,
	}
}

func PrintCityStatistics(properties []*Property, cityNumber int) {
	fmt.Printf("Cidade# %d:\n", cityNumber)

	propertyStats := ""
	sumConsumption := 0
	sumResidents := 0

	for _, property := range properties {
		propertyStats += fmt.Sprintf("%d-%d ", property.Residents, int(property.Consumption/property.Residents))
		sumConsumption += property.Consumption
		sumResidents += property.Residents
	}
	propertyStats = propertyStats[:len(propertyStats)-1] + "\n"

	fmt.Print(propertyStats)
	fmt.Printf("Consumo medio: %.2f m3.\n", math.Floor(float64(sumConsumption)/float64(sumResidents)))
}

func main() {
	var nbProperties int

	cityNumber := 1

	for {
		fmt.Scanln(&nbProperties)

		if nbProperties < 1 {
			break
		}

		properties := make([]*Property, nbProperties)

		for i := 0; i < nbProperties; i++ {
			var residents, consumption int
			fmt.Scanln(&residents, &consumption)
			properties[i] = NewProperty(residents, consumption)
		}

		sort.Slice(properties, func(i, j int) bool {
			return properties[i].Consumption/properties[i].Residents > properties[j].Consumption/properties[j].Residents
		})

		PrintCityStatistics(properties, cityNumber)

		cityNumber++
	}
}
